packages/@ama-sdk/schematics/schematics/ng-update/typescript/v10.3/update-openapiversion.ts

```diff
@@ -0,0 +1,76 @@
+import type {Rule} from '@angular-devkit/schematics';
+import * as semver from 'semver';
+import {getWorkspaceConfig, globInTree} from '@o3r/schematics';
+import {chain} from '@angular-devkit/schematics';
+
+/**
+ * Update open api version used in the project to match with the one used in @ama-sdk/schematics:typescript-core
+ */
+export const updateOpenApiVersionInProject = (): Rule => {
+  const overwriteOpenApiVersion = (pathOfWorkspace: string): Rule => {
+    return (tree, context) => {
+      const packageJson = tree.readText(`${pathOfWorkspace}/package.json`);
+      const amaSdkTypescriptUsage = packageJson.match(/@ama-sdk\/schematics:typescript-core.*",/mg);
+      if (!amaSdkTypescriptUsage) {
+        context.logger.info(`Skipping ng-update for ${pathOfWorkspace} as it does not use `);
+        return tree;
+      }
+      const openapitoolsPaths: string[] = [];
+      amaSdkTypescriptUsage.forEach((script) => {
+        const openapiToolsPath = new RegExp(/@ama-sdk\/schematics:typescript-core.* --spec-config-path[= ]["']?(?:\.\/)?([a-zA-Z-/0-9]*\.json)/).exec(script);
+        if (openapiToolsPath?.length === 2) {
+          openapitoolsPaths.push(`${pathOfWorkspace}/${openapiToolsPath[1]}`);
+        }
+      });
+      if (amaSdkTypescriptUsage.length > openapitoolsPaths.length) {
+        openapitoolsPaths.push(`${pathOfWorkspace}/openapitools.json`);
+      }
+      context.logger.info(`${openapitoolsPaths.length} openapi configuration files found. The following files will be updated to target open api tools version 7.4:
+      ${JSON.stringify(openapitoolsPaths, null, 2)}`);
+      (openapitoolsPaths || []).forEach((path) => {
+        if (!tree.exists(path)) {
+          context.logger.warn(`Missing ${path} file described in the package json.`);
+          return;
+        }
+        const openApiToolsConfig = tree.readJson(path) as any;
+        openApiToolsConfig['generator-cli'] ||= {};
+        if (!openApiToolsConfig['generator-cli'].version || !semver.satisfies('~7.4.0', openApiToolsConfig['generator-cli'].version)) {
+          openApiToolsConfig['generator-cli'].version = '7.4.0';
+        }
+        tree.overwrite(path, JSON.stringify(openApiToolsConfig, null, 2));
+      }
+      );
+      return tree;
+    };
+  };
+
+  return (tree, context) => {
+    let pathsPackageJson: string[];
```

Comment:
```suggestion
    let pathsPackageJson: string[] = ['.'];
```

to be aligned with the type

Reply:
Do we want to run it at the root if we detect workspaces?

Reply:
Then maybe `[]`

Reply:
If you prefer, but you will still need to specify the string[] and you can't do the ||= later on, but push the '.' inside a if(pathsPackageJson.length === 0). Either way, I don't mind

Reply:
In fact, after more thinking and discussion, it seems it is best to actually test the root folder as the openapitools config could be set at the root for a monorepo with different setup for each workspace.